pkg/dc2deployment/dc2deployment.go

```diff
@@ -96,13 +87,63 @@ func Mutate(pluginName string, dc dcAPI.DeploymentConfig, log *logrus.Logger) (d
 			deploy.Spec.Selector.MatchLabels[index] = element
 		}
 	}
-	log.Debugf("[%s] d.Selector: %#v", pluginName, deploy.Spec.Selector)
+	log.Debugf("[%s] Strategy: %#v", pluginName, dc.Spec.Strategy)
+
+	// openshift supoorts Strategy.Type={"Recreate","Custom","Rolling"}
+	// K8s native supports k8s.Strategy.Type={"Recreate","RollingUpdate"}
+	// Custom strategy.type is not supported in K8s native, hence defaulting it to RollingUpdate
+	if dc.Spec.Strategy.Type != "" {
+		if dc.Spec.Strategy.Type == "Rolling" || dc.Spec.Strategy.Type == "Custom" {
+			deploy.Spec.Strategy.Type = deployAPI.DeploymentStrategyType("RollingUpdate")
+		} else {
+			deploy.Spec.Strategy.Type = deployAPI.DeploymentStrategyType(dc.Spec.Strategy.Type)
+		}
+
+	}
+	//Rollingupdate.MaxSurge and Rollingupdate.MaxUnavailable supported only when Strategy type is "RollingUpdate"
+	//If openshift Strategy.type is Custom Rollingupdate.MaxSurge and Rollingupdate.MaxUnavailable is not applicable.
+	if deploy.Spec.Strategy.Type == "RollingUpdate" && dc.Spec.Strategy.Type != "Custom" {
+		deploy.Spec.Strategy.RollingUpdate = new(deployAPI.RollingUpdateDeployment)
+		deploy.Spec.Strategy.RollingUpdate.MaxSurge = dc.Spec.Strategy.RollingParams.MaxSurge
+		deploy.Spec.Strategy.RollingUpdate.MaxUnavailable = dc.Spec.Strategy.RollingParams.MaxUnavailable
+	}
+	deploy.Spec.RevisionHistoryLimit = dc.Spec.RevisionHistoryLimit
 
 	if dc.Spec.Template != nil {
 		dc.Spec.Template.DeepCopyInto(&deploy.Spec.Template)
 	}
-
-	log.Tracef("[%s] deployment %#v", pluginName, deploy)
-
+	// End of Spec Section
+
+	//Status Section Start
+	deploy.Status.AvailableReplicas = dc.Status.AvailableReplicas
+	deploy.Status.ObservedGeneration = dc.Status.ObservedGeneration
+	deploy.Status.Replicas = dc.Status.Replicas
+	deploy.Status.UnavailableReplicas = dc.Status.UnavailableReplicas
+	deploy.Status.UpdatedReplicas = dc.Status.UpdatedReplicas
+
+	if dc.Status.Conditions != nil {
+		deploy.Status.Conditions = make([](deployAPI.DeploymentCondition), len(dc.Status.Conditions))
+
+		for i := range dc.Status.Conditions {
+
+			deploy.Status.Conditions[i].Type = deployAPI.DeploymentConditionType(dc.Status.Conditions[i].Type)
+			deploy.Status.Conditions[i].Status = dc.Status.Conditions[i].Status
+			deploy.Status.Conditions[i].LastUpdateTime = dc.Status.Conditions[i].LastUpdateTime
+			deploy.Status.Conditions[i].LastTransitionTime = dc.Status.Conditions[i].LastTransitionTime
+			deploy.Status.Conditions[i].Reason = dc.Status.Conditions[i].Reason
+			deploy.Status.Conditions[i].Message = dc.Status.Conditions[i].Message
+		}
+	}
+	// End of Status Section
+
+	//Logging the unsupported fields
```

Comment:
My preference is to annotate on the new object only if the original object had the unsupported field set. There are some not-so-popular deploymentConfig settings that might not make to the annotation of the httpproxy.

Reply:
Removed unwanted annotations, retaining only test, trigger,and activeDeadlineSeconds.